### PR TITLE
[AF-2626] - Create Dashbuilder Runtime Image

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: dashbuilder/branch-overrides.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - name: jboss-eap-7-image
+      git:
+        url: https://github.com/jboss-container-images/jboss-eap-7-image.git
+        ref: EAP_733_CR2
+    - name: wildfly-cekit-modules
+      git:
+        url: https://github.com/wildfly/wildfly-cekit-modules.git
+        ref: 0.18.6
+    - name: rhpam-7-image
+      path: /data/dev/sources/rhpam-7-image
```
Comment (Kaitou786) [inline dashbuilder/branch-overrides.yaml:27]: Why is the path given here, instead of the git url?

Comment (spolti) [reply dashbuilder/branch-overrides.yaml:27]: draft yet, as the all other changes are local yet, need to use it, will update when the PR is ready.

Comment (Kaitou786) [reply dashbuilder/branch-overrides.yaml:27]: Sorry, didn't notice :p. Will check when it is ready :)

**Comment by spolti** [comment]
depends on https://github.com/jboss-container-images/jboss-kie-modules/pull/402

**Comment by spolti** [comment]
@jesuino Can you please review as well?

File: dashbuilder/branch-overrides.yaml (hunk 1 above)
Comment (spolti) [reply dashbuilder/branch-overrides.yaml]: @Kaitou786 now it is ready for review :)

**Review by msmagnanijr — DISMISSED** [review]
lgtm

File: dashbuilder/image.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    description: "Boolean property that when true enables external components. Sets the dashbuilder.components.enable property. Default value is false"
+  - name: "DASHBUILDER_EXTERNAL_COMP_DIR"
+    example: "/tmp/test"
+    description: "Configures the base directory for dashbuilder components. Sets the dashbuilder.components.dir property. Default value is /opt/kie/components"
+  - name: "DASHBUILDER_CONFIG_MAP_PROPS"
+    example: "/data/config.properties"
+    description: "Properties file with Dashbuilder configurations, if set, uniq properties will be appended and, if a property is set mode than once, the one from this property file will be used."
+  - name: "KIESERVER_DATASETS"
+    example: "dataset_1,dataset_2"
+    description: "Configures the KIE Server Datasets prefix. For multi datasets, use comma separated values. The valid values for this prefix are: ${datasetPrefix}_LOCATION, ${datasetPrefix}_USER, ${datasetPrefix}_PASSWORD, ${datasetPrefix}_TOKEN. For more information about these properties please take a look on https://docs.jboss.org/jbpm/release/7.45.0.Final/jbpm-docs/html_single/#_kie_server_integration"
```
Comment (jesuino) [inline dashbuilder/image.yaml:69]: I see it is missing `replace_query` - which is used to have dashbuilder creating and replacing queries on Kie Server. Default value is false.

https://github.com/kiegroup/appformer/blob/master/dashbuilder/dashbuilder-backend/dashbuilder-kie-server-backend/src/main/java/org/dashbuilder/kieserver/backend/KieServerConnectionInfoProviderImpl.java#L62

File: dashbuilder/image.yaml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  - name: "KIESERVER_SERVER_TEMPLATES"
+    example: "server_template_1,serverTemplate2"
+    description: "Configures the KIE Server Server Template prefix. For multi serverTemplate, use comma separated values. The valid values for this prefix are: ${serverTemplatePrefix}_LOCATION, ${serverTemplatePrefix}_USER, ${serverTemplatePrefix}_PASSWORD, ${serverTemplatePrefix}_TOKEN. For more information about these properties please take a look on https://docs.jboss.org/jbpm/release/7.45.0.Final/jbpm-docs/html_single/#_kie_server_integration"
```
Comment (jesuino) [inline dashbuilder/image.yaml:72]: I see it is missing `replace_query` - which is used to have dashbuilder creating and replacing queries on Kie Server. Default value is false.

https://github.com/kiegroup/appformer/blob/master/dashbuilder/dashbuilder-backend/dashbuilder-kie-server-backend/src/main/java/org/dashbuilder/kieserver/backend/KieServerConnectionInfoProviderImpl.java#L62

**Review by jesuino — CHANGES_REQUESTED** [review]
Looks good to me, thanks @spolti 

One small change is that we also have a flag for kie server configuration, which is replace_query:

https://github.com/kiegroup/appformer/blob/master/dashbuilder/dashbuilder-backend/dashbuilder-kie-server-backend/src/main/java/org/dashbuilder/kieserver/backend/KieServerConnectionInfoProviderImpl.java#L62

It is a boolean property which when true will make Dashbuilder create attempt to update the query on Kie Server. 


The valid values for this prefix are: ${serverTemplatePrefix}_LOCATION, ${serverTemplatePrefix}_USER, ${serverTemplatePrefix}_PASSWORD, ${serverTemplatePrefix}_TOKEN

**Comment by spolti** [comment]
@jesuino updated, will be updating jboss-kie-modules in a bit.

**Review by jesuino — APPROVED** [review]
LGTM

**Review by Kaitou786 — APPROVED** [review]

**Review by msmagnanijr — APPROVED** [review]
lgtm

**Review by jakubschwan — APPROVED** [review]

**Pull request merged**